src/main.c

```diff
@@ -807,11 +822,11 @@ static int listener_setup_ssl(h2o_configurator_command_t *cmd, h2o_configurator_
     if (h2o_configurator_parse_mapping(cmd, *ssl_node, "key-file:s",
                                        "certificate-file:s,raw-pubkey-file:s,min-version:s,minimum-version:s,max-version:s,"
                                        "maximum-version:s,cipher-suite:s,ocsp-update-cmd:s,ocsp-update-interval:*,"
-                                       "ocsp-max-failures:*,dh-file:s,cipher-preference:*,neverbleed:*,http2-origin-frame:*",
+                                       "ocsp-max-failures:*,dh-file:s,cipher-preference:*,neverbleed:*,http2-origin-frame:*,client-verify:*,client-CA-file:s",
                                        &key_file, &certificate_file, &raw_pubkey_file, &min_version, &min_version, &max_version,
                                        &max_version, &cipher_suite, &ocsp_update_cmd, &ocsp_update_interval_node,
                                        &ocsp_max_failures_node, &dh_file, &cipher_preference_node, &neverbleed_node,
-                                       &http2_origin_frame_node) != 0)
+                                       &http2_origin_frame_node, &client_verify_node, &client_CA_file) != 0)
```

Comment:
Do we want to support mTLS using the default certificate store? Assuming that we do not want to, I need only one configuration knob that specifies the CA file. If that CA file is specified, we would turn on mTLS.

Reply:
For mTLS, I think that there are 2 use cases: 
1. user provides private Root CA file or bundle. It is located and specified in "client_CA_file".
2. public Root CA, it should be included in default trust store. So "client_CA_file" can be null or blank. (need to test this case though)

Reply:
Need to handle "client_CA_file==NULL" case, use default Root CA trust store.

Reply:
Thank you for elaborating. Generally speaking, I tend to agree that there are two use cases.

However, to support the latter use case (of using public root as trust anchor) we need a way of signalling the web applications the identity being authenticated, so that the applications can authorize the users. In contrast to application-specific trust anchors provide both authentication and authorization, public roots only provide authentication. There is a [de-facto standard](https://httpd.apache.org/docs/2.4/mod/mod_ssl.html#page-header) regarding how the information carried by client certificates should be communicated to the web application, but we have not implemented that in this PR.

To paraphrase, this PR does not cover the latter use case at the moment. Therefore, I tend to wonder if (and how) we could start with supporting on the former use-case.

Reply:
Agree. use single flag as suggested. Will revisit if there are additional use cases and requirements.